### PR TITLE
Issue 239: Voice buttons and TTS buttons

### PR DESCRIPTION
**Pull request by MegaGeese**
#239 

-Added ability for button based questions to have their buttons read by TTS
-Added check for if button responses are buttons or images.
-Added alert to notify user if their configuration is not supported by the selected TDF
-Bugfix: Fixed bug where questions were not being read if audio prompt mode was set to just "question"

### CODE REVIEW

File: mofacts/client/views/experiment/card.js [hunk 1]
```diff
@@ -1973,11 +1973,18 @@ function beginQuestionAndInitiateUserInput(delayMs, deliveryParams) {
       });
     }, timeuntilaudio);
   } else { // Not a sound - can unlock now for data entry now
-    const questionToSpeak = currentDisplay.text || currentDisplay.clozeText;
+    const questionToSpeak = currentDisplay.clozeText || currentDisplay.text;
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/card.js:1976]: Why did you switch these?

File: mofacts/client/views/home/profile.js [hunk 2]
```diff
@@ -427,7 +427,7 @@ async function selectTdf(currentTdfId, lessonName, currentStimuliSetId, ignoreOu
 
   // Check to see if the user has turned on audio prompt.
   // If so and if the tdf has it enabled then turn on, otherwise we won't do anything
-  const userAudioPromptFeedbackToggled = (audioPromptFeedbackView == 'feedback') || (audioPromptFeedbackView == 'all');
+  const userAudioPromptFeedbackToggled = (audioPromptFeedbackView == 'feedback') || (audioPromptFeedbackView == 'all') || (audioPromptFeedbackView == 'question');
```
Comment (andrewtackett) [inline mofacts/client/views/home/profile.js:430]: Ideally, this would be a separate commit as it's solving a different issue, though for now this is fine.

File: mofacts/client/views/home/profile.js [hunk 3]
```diff
@@ -488,6 +488,18 @@ async function selectTdf(currentTdfId, lessonName, currentStimuliSetId, ignoreOu
     };
     await updateExperimentState(newExperimentState, 'profile.selectTdf');
 
+    const curTdf = await meteorCallAsync('getTdfById', currentTdfId);
```
Comment (andrewtackett) [inline mofacts/client/views/home/profile.js:491]: There's actually already a function for this: https://github.com/memphis-iis/mofacts-ies/blob/postgresMigration/mofacts/client/views/experiment/card.js#L864

**Review by andrewtackett — CHANGES_REQUESTED** [review]
Go ahead and switch out the explicit checking for images to use the pre-existing function instead and it should be good to go.

File: mofacts/client/views/experiment/card.js (hunk 1 above)
Comment (MegaGeese) [reply mofacts/client/views/experiment/card.js:1976]: In the testStim.json file the question is stored in clozeText instead of text. So the API was reading text while clozeText was appearing on the screen.

**Review by andrewtackett — APPROVED** [review]
Looks good.
